viz_scripts/scaffolding.py

```diff
@@ -105,6 +115,30 @@ def expand_userinputs(labeled_ct):
     disp.display(expanded_ct.head())
     return expanded_ct
 
+def expand_inferredlabels(inferred_ct):
+    if len(inferred_ct) == 0:
+        return inferred_ct
+
+    max_labels_list = []
+    for _, row in inferred_ct.iterrows():
+        # In the trip, prioritize availabilty of user_input over inferred_labels for label selection
+        if row.user_input == {}:
+            # Extract the label which has highest "p" value
+            max_entry = max(row.inferred_labels, key=lambda x: x['p'])
+            if (max_entry['p'] > row.confidence_threshold):
+                max_labels_list.append(max_entry['labels'])
+            else:
+                max_labels_list.append({})
```

Comment:
So if there are no user labels and the confidence is too low there won't be any labels? Do these entries get filtered out later?

Reply:
Whenever there is no user labels and the 'p' value for the inferred_labels is less than the confidence_threshold. We are not considering the inferred labels of mode_confirm, purpose_confirm and replaced_mode, therefore an empty dictionary is being added. 
Later these columns would be converted as NaN for mode_confirm, purpose_confirm and replaced_mode while creating a dataframe, which is subsequently added to the original dataframe.

When I looked up for 
```
(expanded_ct_inferred['mode_confirm'].value_counts(dropna=False), expanded_ct_inferred['Mode_confirm'].value_counts(dropna=False))
```
I observed the below:

mode_confirm:

```

e-bike               1868
 drove_alone          1101
 shared_ride           768
 NaN                   655
 walk                  467
 atv                    42
 e_car_shared_ride      15
 e_car_drove_alone      13
 not_a_trip             12
 bike                    8
 side_by side            6
 free_shuttle            3
 lawn_mower              2
                         2
 moco_transit            2
 air                     2
 atv_ride                1
 Name: mode_confirm, dtype: int64

```

Mode_confirm:

```
E-bike                  1868
 Gas Car, drove alone    1101
 Gas Car, with others     768
 Other                    710
 Walk                     467
 E-car, with others        15
 E-car, drove alone        13
 Not a Trip                12
 Regular Bike               8
 Free Shuttle               3
 Airplane                   2
 Name: Mode_confirm, dtype: int64)
```

The values processed as `NaN` was eventually converted to `Other`, which is wrong. 

> Do these entries get filtered out later?

I have to filter it.

Reply:
Considered the case of filtering with `NaN` values, but there could be some existing `NaN` values apart from the way it was computed earlier. Decided to pass `uncertain` values for the keys to `mode_confirm`, `purpose_confirm` and `replaced_mode` in the dictionary, and add it to the list.
After the convergence with the original dataframe, filter out the rows which have these mode_confirm, purpose_confirm and replaced_mode as `uncertain`.

Reply:
I am fine with this for now, but I would expect to revisit and discuss as part of/after the footprint calculations. If the user labels are not present, they are mapped to NaN and we drop them (`dropna`). Why do we want to handle this differently? (i.e. why don't we just `dropna` again?) Given that we already indicate that we only represent a subset of the trips, why do we also need to label some of those as `uncertain`?